src/lib389/lib389/cli_conf/__init__.py

```diff
@@ -116,11 +118,30 @@ def generic_object_edit(dsldap_object, log, args, arg_to_attr):
                 value = [value]
             if not (attr in existing_attributes and value == ensure_list_str(existing_attributes[attr])):
                 modlist.append((ldap.MOD_REPLACE, attr, value))
+            else:
+                # Track attributes that didn't change for better error messages
+                unchanged_attrs[attr] = value
+
     if len(modlist) > 0:
         dsldap_object.apply_mods(modlist)
         log.info("Successfully changed the %s", dsldap_object.dn)
     else:
-        raise ValueError("There is nothing to set in the %s plugin entry" % dsldap_object.dn)
+        # Check if we're trying to enable/disable a plugin and it's already in that state
+        enabled_attr = None
+        for arg, attr in arg_to_attr.items():
+            if arg == 'enabled' and attr in unchanged_attrs:
```

Comment:
Iterating on dict items then testing equality on the key looks pretty unusual.
Why not getting directly the value from the key: (using get() or []) as usual ?
  attr = arg_to_attr.get('enabled', None)
  if attr in unchanged_attrs:
                enabled_attr = attr
  or:
    with contextlib.suppress(KeyError):
        if arg_to_attr['enabled'] in unchanged_attrs:
            enabled_attr =  arg_to_attr['enabled']

Reply:
Good point! Fixed